code/modules/GOAP/monkey/actions.dm

```diff
@@ -0,0 +1,351 @@
+	cost = 2
+
+/datum/goap_action/monkey/GetItem/New()
+	..()
+	preconditions = list()
+	preconditions["hasItem"] = FALSE
+	effects = list()
+	effects["hasItem"] = TRUE
+
+/datum/goap_action/monkey/GetItem/AdvancedPreconditions(mob/living/carbon/monkey/C, list/worldstate)
+	if(C.IsDeadOrIncap() || C.restrained())
+		return FALSE
+	for(var/obj/item/I in oview(4, C))
+		if(I && !C.blacklistItems[I] && !HAS_TRAIT(I, TRAIT_NODROP))
+			if(I.force > C.best_force)
+				C.best_force = I.force
+				target = I
+	return (target != null)
+
+/datum/goap_action/monkey/GetItem/RequiresInRange(atom/agent)
+	return TRUE
+
+/datum/goap_action/monkey/GetItem/Perform(mob/living/carbon/monkey/C)
+	C.drop_all_held_items() // who cares about these items, i want that one!
+	C.equip_item(target)
+	action_done = TRUE
+	return ..()
+
+/datum/goap_action/monkey/GetItem/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/monkey/pickpocket
+	name = "Pickpocket Item"
+	cost = 1
+
+/datum/goap_action/monkey/pickpocket/New()
+	..()
+	preconditions = list()
+	preconditions["hasHumanItem"] = FALSE
+	preconditions["enemyAttack"] = FALSE
+	effects = list()
+	effects["hasHumanItem"] = TRUE
+
+/datum/goap_action/monkey/pickpocket/AdvancedPreconditions(mob/living/carbon/monkey/C, list/worldstate)
+	if(C.IsDeadOrIncap() || C.restrained())
+		return FALSE
+	for(var/mob/living/carbon/human/H in oview(4, C))
+		if(H)
+			for(var/obj/item/I in H.held_items)
+				if(I && !C.blacklistItems[I] && I.force > C.best_force)
+					target = H
+					C.pickupTarget = I
+					break
+	return (C.pickupTarget != null)
+
+/datum/goap_action/monkey/pickpocket/RequiresInRange(atom/agent)
+	return TRUE
+
+/datum/goap_action/monkey/pickpocket/Perform(mob/living/carbon/monkey/C)
+	C.drop_all_held_items() // who cares about these items, i want that one!
+	var/mob/M = target
+	M.visible_message("<span class='warning'>[C] starts trying to take [C.pickupTarget] from [M]!</span>", "<span class='danger'>[C] tries to take [C.pickupTarget]!</span>")
+	C.pickpocket(M)
+	action_done = TRUE
+	return ..()
+
+/datum/goap_action/monkey/pickpocket/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/monkey/disarm
+	name = "Disarm"
+	cost = 2 // don't disarm if you can grab a weapon and attack with it
+	cooldown = 20
+
+/datum/goap_action/monkey/disarm/New()
+	. = ..()
+	preconditions = list()
+	preconditions["attackEnemy"] = FALSE
+	preconditions["grabbedWeapon"] = FALSE
+	preconditions["disarmAvailable"] = TRUE
+	effects = list()
+	effects["attackEnemy"] = TRUE
+
+/datum/goap_action/monkey/disarm/AdvancedPreconditions(mob/living/carbon/monkey/C, list/worldstate)
+	if(length(C.enemies))
+		var/list/around = view(MONKEY_ENEMY_VISION, C)
+		for(var/mob/living/L in around)
+			if(C.should_target(L))
+				if(!L.stat && L.held_items) // conscious
+					target = L
+					for(var/obj/item/I in L.held_items)
+						if(!(I.item_flags & ABSTRACT))
+							C.pickupTarget = I
+					break
+	return (target != null)
```

Comment:
haha bro i heard u liked indentation

Reply:
for real tho this could use some early returns

Reply:
where should I return here?

Reply:
Probably something like
```
	if(!length(C.enemies))
		return FALSE
	var/list/around = view(MONKEY_ENEMY_VISION, C)
	for(var/mob/living/L in around)
		if(!C.should_target(L))
			continue
		if(L.stat || !L.held_items) // conscious
			continue
		target = L
		for(var/obj/item/I in L.held_items)
			if(!(I.item_flags & ABSTRACT))
				C.pickupTarget = I
		break
	return (target != null)
```